internal/backend/triggers/triggers.go

```diff
@@ -28,20 +31,75 @@ func (m *triggers) Create(ctx context.Context, trigger sdktypes.Trigger) (sdktyp
 
 	trigger = trigger.WithNewID()
 
+	if trigger.ConnectionID() == fixtures.BuiltinSchedulerConnectionID {
+		if schedule, _ := trigger.Data()[sdktypes.ScheduleExpression].ToString(); schedule != "" {
+			scheduleID := newScheduleID(trigger)
+			if err := m.scheduler.Create(ctx, scheduleID, schedule, trigger.ID()); err != nil {
+				return sdktypes.InvalidTriggerID, err
+			}
+			trigger = trigger.WithUpdatedData(sdktypes.ScheduleIDKey, sdktypes.NewStringValue(scheduleID))
+		}
+	}
+
 	if err := m.db.CreateTrigger(ctx, trigger); err != nil {
 		return sdktypes.InvalidTriggerID, err
 	}
-
 	return trigger.ID(), nil
 }
 
 func (m *triggers) Update(ctx context.Context, trigger sdktypes.Trigger) error {
+	prevTrigger, err := m.db.GetTrigger(ctx, trigger.ID())
+	if err != nil {
+		return err
+	}
+
+	prevData := prevTrigger.Data()
+	prevScheduleVal, isSchedulerPrevTrigger := prevData[sdktypes.ScheduleExpression]
+	scheduleID, _ := prevData[sdktypes.ScheduleIDKey].ToString()
```

Comment:
why isn't the schedule id just the trigger id?

Reply:
readability for the user through temporal UI. `daily_trg_xxx` IMO is better then just `trg_xxx` which explains nothing.
If we don't care we could remove this part that creates and saves scheduleID in the data section of the trigger and assume permanently that scheduleID == triggerID

Reply:
done